### PR TITLE
[RateLimiter] fix probably undefined variable $expireAt

### PR DESCRIPTION
**Pull request by nommyde**
$expireAt is undefined when $limiterState->getExpirationTime() returns null. Built in limiters don't return null, but interface allows to do it. I've made my custom limiter with unexpired state and found this problem

| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| License       | MIT

### CODE REVIEW

File: src/Symfony/Component/RateLimiter/Storage/InMemoryStorage.php [hunk 1]
```diff
@@ -57,4 +50,15 @@ public function delete(string $limiterStateId): void
 
         unset($this->buckets[$limiterStateId]);
     }
+
+    private function getExpireAt(LimiterStateInterface $limiterState): ?float
+    {
+        if (isset($this->buckets[$limiterState->getId()])) {
+            return $this->buckets[$limiterState->getId()][0];
```
Comment (stof) [inline src/Symfony/Component/RateLimiter/Storage/InMemoryStorage.php]: This is changing the existing logic, where `$limiterState->getExpirationTime()` wins over the existing bucket expiration time.

**Review by stof — CHANGES_REQUESTED** [review]

**Comment by stof** [comment]
Instead of adding a new method, I would just add a `$expireAt  = null` before the first `if`

**Comment by nommyde** [comment]
> Instead of adding a new method, I would just add a `$expireAt = null` before the first `if`

In my view this approach is the reason of this bug. So I've decided to change it. It leads to such cases like propbably undefined variables, extra computations and priority confusions.

**Comment by nicolas-grekas** [comment]
Thank you @Nommyde.

**Pull request merged**